lib/connection.js

```diff
@@ -145,32 +143,56 @@ class Connection extends EventEmitter {
     }
 
     let message;
+    let isNewSession = true;
+    const handshakePayload = new Array(2);
```

Comment:
If the whole purpose of this array is to eventually unpack it using spread operator, I'd actually find it more readable to use just two named variables (e.g., `handshakeStrategy` and `handshakeCredentials`).